### PR TITLE
Increase code coverage

### PR DESCRIPTION
**Pull request by TuxedoFish**
- adds util/ test files (Hitting untested parts and still needs future work)
- improve toHaveElementProperty tests
- Change elementsUtil forEach to for loop as seemed to not be functioning correctly
- update options test to increase coverage
- remove jasmineUtil from test coverage since it was a file extracted out of jasmine 2.5.2 and seems to thererfore be outside of scope of increasing the coverage for the repo, this was my thinking but what do you think?

All these things considered statement coverage is now just over 99% but branch coverage needs improvement and there are still missing test cases I think

### CODE REVIEW

File: src/util/elementsUtil.ts [hunk 1]
```diff
@@ -40,7 +40,9 @@ export const updateElementsArray = (
 
         // add every refetched item to original element array
         if (received.length === 0) {
-            refetched.forEach(el => received.push(el))
+            for(var i=0; i<refetched.length; i++) {
```
Comment (christian-bromann) [inline src/util/elementsUtil.ts:43]:
```suggestion
            for(let i=0; i < refetched.length; i++) {
```

File: test/util/executeCommand.test.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { waitUntil } from '../../src/utils'
+
+describe('refetchElements', () => {
+    test('is array', async () => {
+        const els = await $$('sel') 
+        // @ts-ignore
+        els.parent._length = 5;
+        // Passing in array should cause elements
+        // to be refetched hence length should be 5
+        let condition = jest.fn(() => { return { result: true, value: "Test Value "} } )
```
Comment (christian-bromann) [inline test/util/executeCommand.test.ts:11]:
```suggestion
        const condition = jest.fn(() => { return { result: true, value: "Test Value "} } )
```

File: test/util/executeCommand.test.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        const result = await executeCommand.call(this, els, condition, {}, ["Test Property", "Test Value"], true)
+        expect(condition).toHaveBeenCalledTimes(5)
+    })
+
+    test('empty array', async () => {
+        const els = await $$('sel') 
+        while (els.length > 0) { els.pop() }
+        // Passing in array should cause elements
+        // to be refetched hence length should be 5
+        let condition = jest.fn(() => { return { result: true, value: "Test Value "} } )
```
Comment (christian-bromann) [inline test/util/executeCommand.test.ts:23]:
```suggestion
        const condition = jest.fn(() => { return { result: true, value: "Test Value "} } )
```

File: test/util/formatMessage.test.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                    { message: '', containing: false }
+                )
+            })
+
+            test('starting message', () => {
+                expect(actual).toMatch('Expect Test Subject to Test Verb Test Expectation')
+            })
+
+            test('diff string', () => {
+                let diffString = printDiffOrStringify('Test Expected', 'Test Actual', 'Expected', 'Received', true)
```
Comment (christian-bromann) [inline test/util/formatMessage.test.ts:28]:
```suggestion
                const diffString = printDiffOrStringify('Test Expected', 'Test Actual', 'Expected', 'Received', true)
```

File: test/util/formatMessage.test.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+                        { message: '', containing: false }
+                    )
+                })
+
+                test('starting message', () => {
+                    expect(actual).toMatch("Expect Test Subject not to Test Verb Test Expectation")
+                })
+
+                test('diff string', () => {
+                    let diffString = printDiffOrStringify('Test Expected', 'Test Actual', 'Expected [not]', 'Received      ', true)
```
Comment (christian-bromann) [inline test/util/formatMessage.test.ts:55]:
```suggestion
                    const diffString = printDiffOrStringify('Test Expected', 'Test Actual', 'Expected [not]', 'Received      ', true)
```

File: test/util/formatMessage.test.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+                        { message: '', containing: false }
+                    )
+                })
+
+                test('starting message', () => {
+                    expect(actual).toMatch("Expect Test Subject not to Test Verb Test Expectation")
+                })
+
+                test('diff string', () => {
+                    let diffString = `Expected [not]: ${printExpected("Test Same")}\n` +
```
Comment (christian-bromann) [inline test/util/formatMessage.test.ts:79]:
```suggestion
                    const diffString = `Expected [not]: ${printExpected("Test Same")}\n` +
```

File: test/util/formatMessage.test.ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+                    { message: '', containing: true }
+                )
+            })
+
+            test('starting message', () => {
+                expect(actual).toMatch('Expect Test Subject to Test Verb Test Expectation containing')
+            })
+
+            test('diff string', () => {
+                let diffString = printDiffOrStringify('Test Expected', 'Test Actual', 'Expected', 'Received', true)
```
Comment (christian-bromann) [inline test/util/formatMessage.test.ts:108]:
```suggestion
                const diffString = printDiffOrStringify('Test Expected', 'Test Actual', 'Expected', 'Received', true)
```

File: test/util/formatMessage.test.ts [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+                    { message: 'Test Message', containing: false }
+                )
+            })
+
+            test('starting message', () => {
+                expect(actual).toMatch('Test Message\nExpect Test Subject to Test Verb Test Expectation')
+            })
+
+            test('diff string', () => {
+                let diffString = printDiffOrStringify('Test Expected', 'Test Actual', 'Expected', 'Received', true)
```
Comment (christian-bromann) [inline test/util/formatMessage.test.ts:134]:
```suggestion
                const diffString = printDiffOrStringify('Test Expected', 'Test Actual', 'Expected', 'Received', true)
```

File: test/util/formatMessage.test.ts [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+                    { message: 'Test Message', containing: false }
+                )
+            })
+
+            test('starting message', () => {
+                expect(actual).toMatch('Expect Test Subject to Test Verb Test Expectation Test Arg2')
+            })
+
+            test('diff string', () => {
+                let diffString = printDiffOrStringify('Test Expected', 'Test Actual', 'Expected', 'Received', true)
```
Comment (christian-bromann) [inline test/util/formatMessage.test.ts:160]:
```suggestion
                const diffString = printDiffOrStringify('Test Expected', 'Test Actual', 'Expected', 'Received', true)
```

**Review by christian-bromann — COMMENTED** [review]
Great PR! Thanks a lot. Some minor comments.

Do you know why some of the tests are failing?`

**Comment by TuxedoFish** [comment]
Ahh oversight from me, I was running the tests in isolation and completely missed this!! Thanks for the comments I will fix those things and get the tests passing cheers!

**Review by christian-bromann — APPROVED** [review]
This is perfect, thanks so much 🙏

**Pull request merged**